nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/FTPTransfer.java

```diff
@@ -321,13 +325,22 @@ public FlowFile getRemoteFile(final String remoteFileName, final FlowFile origFl
         FlowFile resultFlowFile;
         try (InputStream in = client.retrieveFileStream(remoteFileName)) {
             if (in == null) {
-                final String response = client.getReplyString();
-                // FTPClient doesn't throw exception if file not found.
-                // Instead, response string will contain: "550 Can't open <absolute_path>: No such file or directory"
-                if (response != null && response.trim().endsWith("No such file or directory")) {
-                    throw new FileNotFoundException(response);
+                final int replyCode = client.getReplyCode();
+
+                final String reply = client.getReplyString();
+                if (reply == null) {
```

Comment:
I did not review the ftp client code but I'd imagine the reply code is extracted by it from the reply string.  Here we pull the code, then the string, then check if the string is null.  I suspect we want code extraction AFTER the null and/or empty check.

Reply:
Thanks for the feedback @joewitt! After taking a closer look at the commons-net FTP code, it appears that the FTP client will through an exception if the reply string is `null` when attempting to parse the reply code, so will move `getReplyCode()` after `getReplyString()`.